src/pages/batchImports/[resourceId]/index.js

```diff
@@ -0,0 +1,302 @@
+        return parseFloat(value);
+      case 5:
+        return isAPI 
+          ? formatDateForImport(value) 
+          : formatDateDefault(formatDate(value)); 
+      default:
+          return value;
+  }
+}
+
+const parseCSV = (text, columns, setState, refetch) => {
+  const lines = text.split('\n').filter(line => line.trim());
+
+  if (lines.length === 0) return;
+
+  const headers = lines[0].split(',').map(header => header.trim());
+
+  const columnMap = columns.reduce((map, col) => {
+    map[col.headerName] = col;
+
+    return map;
+  }, {});
+
+  const orderedColumns = headers.map(header => columnMap[header]);
+
+  const rows = lines.slice(1).map(line => {
+    const values = line.split(',').map(value => value.trim());
+
+    return orderedColumns.reduce((obj, col, index) => {
+      const header = headers[index];
+      if (col) {
+        obj[col.field] = convertValue(values[index], col.dataType);
+      } else {
+        obj[header] = values[index];
+      }
+
+      return obj;
+    }, {});
+  });
+
+  const dataFromCSV = transform(rows);
+  setState(prevState => ({
+    ...prevState,
+    gridData: dataFromCSV
+  }));
+  refetch();
+};
+
+const getImportData = async (state, columns, access, platformLabels, postRequest, stack, stackError, invalidate) => {
+  const mandatoryColumns = columns.filter(col => col.mandatory);
+
+  const missingFields = state.gridData.list.flatMap(row =>
+    mandatoryColumns
+      .filter(col => row[col.field] === null || row[col.field] === undefined || row[col.field] === '')
+      .map(col => col.headerName)
+  );
+
+  if (missingFields.length > 0) {
+    const uniqueMissingFields = [...new Set(missingFields)];
+    stackError({
+      message: `${uniqueMissingFields.join(', ')} ${uniqueMissingFields.length > 1 ? 'are' : 'is'} mandatory field${uniqueMissingFields.length > 1 ? 's' : ''}.`
+    });
+
+    return;
+  }
+
+  const convertedData = state.gridData.list.map(row => {
+    return Object.keys(row).reduce((acc, key) => {
+      const col = columns.find(c => c.field === key);
+      let value = row[key];
+      value = value === '' ? null : value;
+      acc[key] = col ? convertValue(value, col.dataType, true) : value;
+
+      return acc;
+    }, {});
+  });
+
+  const data = {
+    [state.objectName]: convertedData
+  };
+
+  try {
+    const res = await postRequest({
+      extension: state.endPoint,
+      record: JSON.stringify(data)
+    });
+
+    stack({
+      Component: ProgressForm,
+      props: {
+        recordId: res.recordId,
+        access
+      },
+      width: 500,
+      height: 450,
+      title: platformLabels.Progress
+    });
+    invalidate();
+    toast.success(platformLabels.Imported);
+  } catch (exception) {}
+};
+
+const BatchImports = () => {
+  const { stack } = useWindow()
+  const { stack: stackError } = useError()
+  const router = useRouter()
+
+  const [columns, setColumns] = useState([])
+
+  const [state, setState] = useState({
+    gridData: [],
```

Comment:
it's better to separate the states, because here for example if you need to update only name you will pass on the whole state to retrieve the old data and modify only name